applications/tari_indexer/src/substate_manager.rs

```diff
@@ -251,6 +255,23 @@ impl SubstateManager {
         Ok(nfts)
     }
 
+    pub async fn get_event_from_db(
+        &self,
+        template_address: String,
+        tx_hash: String,
```

Comment:
nit: What are these strings? They can technically be anything. In general, we want to use the compiler to ensure that the database layer can only have well-formed data. How that data is serialized is not a concern of callers. Not doing this results in: 
- malformed data being entered into the database
- Changes to data types may only break things at the highest level (e.g. something in the call to the jrpc broke vs the decoding in this specific db function is broken)
- conversion code littering the codebase (see `EventQuery::get_event`)
- get and update/insert operations can have inconsistencies 
- tight coupling with a specific database and/or ORM (e.g. below `NewEvent`  is a diesel helper struct, we may want to use a non-relational database, we may want to use postgres etc)

```suggestion
        template_address: &TemplateAddress,
        tx_hash: PayloadId, // This is usually a bit muddy,  Hash and FixedHash are also used across the codebase
```

Reply:
Great suggestion ! I did refactor the code accordingly, using the types `TemplateAddress` and `PayloadId` for the signature types. Then made use of the `.to_string()` method to convert to string, for query build in the `fn get_event` method on the substate manager reader.